Remove editor dependency

madmiraal opened the pull request:
Building on #21978. This patch removes the dependency on the editor directory being in the build's include path. The first commit updates the files that were dependent on the editor directory being in the include path. The second commit removes editor from the SConstruct CPPATH.

madmiraal committed:
Remove dependency on the editor directory being in the build's include path.

- Add or remove the necessary subdirectorires to the includes to remove
dependency on the editor directory being in the build's include path.
- Ensure includes in modified files conform to style guideline.
- Remove editor from the build include path.

akien-mga commented:
There's now a style issue as includes should be ordered alphabetically. Otherwise it looks good, but I'd likely squash the two commits together.

madmiraal commented:
I've applied clang-format to each of the modified files. To make this work properly I also removed the include blocks that had been created in editor/plugins/asset_library_editor_plugin.h. And I've squashed the commits.

akien-mga commented:
Thanks!
